### ISSUE TITLE
/etc/hosts on trusty needs to be fixed

### ISSUE THREAD

**Issue by waja** [travis-ci/travis-ci#4978]
/etc/hosts contains "::1" (and other IPv6 related entries). As those hosts don't have IPv6, those needs to be removed.

``` sh
$ cat /etc/hosts; ip addr show; exit 0
127.0.0.1 localhost  trusty64 nettuno travis vagrant ip6-localhost ip6-loopback
127.0.1.1 localhost trusty64 nettuno travis vagrant ip6-localhost ip6-loopback
::1 localhost trusty64 nettuno travis vagrant ip6-localhost ip6-loopback

fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1460 qdisc mq state UP group default qlen 1000
    link/ether 42:01:0a:f0:00:2b brd ff:ff:ff:ff:ff:ff
    inet 10.240.0.43/32 brd 10.240.0.43 scope global eth0
       valid_lft forever preferred_lft forever
3: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default 
    link/ether 02:42:f7:42:51:4a brd ff:ff:ff:ff:ff:ff
    inet 172.17.42.1/16 scope global docker0
       valid_lft forever preferred_lft forever
4: lxcbr0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default 
    link/ether 1a:fe:43:73:0c:1e brd ff:ff:ff:ff:ff:ff
    inet 10.0.3.1/24 brd 10.0.3.255 scope global lxcbr0
       valid_lft forever preferred_lft forever
```

I worked around this by using:

``` sh
- sudo [ $(ip addr show | grep "inet6 ::1" | wc -l) -lt "1" ] && sudo sed -i '/^::1/d' /etc/hosts
```

**Comment by BanzaiMan** [comment]
Thanks for opening the issue. Is the presence of this line causing problems? If so, do you have a build log URL to show that?

**Comment by waja** [comment]
For example https://travis-ci.org/waja/monitoring-plugins/builds/86399407. check_fping is taking "::1" as this resolves from 'localhost'

**Comment by waja** [comment]
You can see fping6 in https://travis-ci.org/waja/monitoring-plugins/builds/86399407#L3553 while we try to ```check_fping localhost```

**Comment by bdarnell** [comment]
Here's another build that fails because `localhost` resolves to an ipv6 address but this address cannot be used: https://travis-ci.org/bdarnell/tornado/jobs/141611571

The trusty etc/hosts file is funky in another way: `localhost` (and a bunch of other names) map to multiple IP addresses on the loopback interface:

```
127.0.0.1 localhost  trusty64 nettuno travis vagrant ip6-localhost ip6-loopback
127.0.1.1 localhost trusty64 nettuno travis vagrant ip6-localhost ip6-loopback
::1 localhost trusty64 nettuno travis vagrant ip6-localhost ip6-loopback
```

This breaks some testing utilities in the released version of Tornado (tornadoweb/tornado#1574). While it's sometimes useful to split local names across different loopback IPs, I know of no reason to give the same names multiple IPs. I think `/etc/hosts` should look like this:

```
127.0.0.1 localhost
127.0.1.1 trusty64 nettuno travis vagrant
```

(or the four hostnames on the second line could be split across four different loopback IPs). It might also be a good idea to add an entry for the name in `/etc/hostname` (something like `testing-gce-e51d2e2e-4719-4ccc-a457-79b5d579b003`) with a distinct loopback IP. (I don't have any need for this, but it's the usual reason for introducing loopback IPs other than 127.0.0.1)
